alica_tests/src/test/test_agent_dies.cpp

```diff
@@ -55,8 +55,8 @@ TEST_F(AlicaEngineAgentDiesTest, AgentIsRemoved)
 
     TestClock* c1 = new TestClock();
     TestClock* c2 = new TestClock();
-    aes[0]->setAlicaClock(std::unique_ptr<AlicaClock>(c1));
-    aes[1]->setAlicaClock(std::unique_ptr<AlicaClock>(c2));
+    acs[0]->setClock<std::unique_ptr<AlicaClock>(c1);
```

Comment:
The TestClocks c1, c2 should be replaced by a call to setClock: `setClock<TestClock>()` and subsequent calls to TestClock::increment() need to be made by downcasting return value of getAlicaClock()